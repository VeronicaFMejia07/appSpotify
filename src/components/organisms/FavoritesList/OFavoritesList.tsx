import React from 'react'
import 'components/organisms/Playlist/OPlaylist.scss';
import { useAppSelector } from 'redux/hooks/hooks';
import { selectFavoritesInfo, setFavorites } from 'redux/slices/favoritesSlice';
import MFavorites from 'components/molecules/Favorite/MFavorites';

//FunciÃ³n que recibe la lista de canciones de favoritos y las recorre por medio del map
const OFavoritesList = () => {
  const {favorites} = useAppSelector(selectFavoritesInfo);//TRAE LA PLAYLIST-SELECTOR
  console.log(favorites)

  return (
    <section className='section__MPlaylist'>
      <article className='section__playlist'>
      {
          favorites.items.map((item, index) => (
            <MFavorites key={index} item={item} />
          ))
        }
      </article>
    </section>
  )
}

export default OFavoritesList