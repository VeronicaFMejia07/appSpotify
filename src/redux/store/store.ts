import { configureStore } from "@reduxjs/toolkit";
import {userSlice} from "redux/slices/userSlice";
import {playlistSlice } from "redux/slices/playlistSlice";
import {favoritesSlice} from "redux/slices/favoritesSlice";

export const store= configureStore({
    reducer:{
        playlist: playlistSlice.reducer,
        favorites: favoritesSlice.reducer,
        user: userSlice.reducer,
    }
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export default store