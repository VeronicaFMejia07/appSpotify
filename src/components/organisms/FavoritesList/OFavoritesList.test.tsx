import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from 'redux/store/store';
import OFavoritesList from './OFavoritesList';

//Describir la prueba que se este haciendo
describe('Pruebas en OFavoritesList', () => {
    test('Renderiza componente--OFavoritesList', () => {
        const { container } = render(
            <Provider store={store}>
                <OFavoritesList />
            </Provider>
        )

        //Evalua que el container sea igual a lo que contiene el componente
        expect(container).toMatchSnapshot();
    })

})


