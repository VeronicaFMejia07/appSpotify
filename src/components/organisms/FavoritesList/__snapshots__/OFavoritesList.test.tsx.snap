// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en OFavoritesList Renderiza componente--OFavoritesList 1`] = `
<div>
  <section
    class="section__MPlaylist"
  >
    <article
      class="section__playlist"
    />
  </section>
</div>
`;
